pkg/operator/status.go

```diff
@@ -203,13 +203,13 @@ func (c *Controller) syncStatus(cr *imageregistryv1.Config, deploy *appsapi.Depl
 		operatorAvailable.Message = "The registry configuration is set to unmanaged mode"
 		operatorAvailable.Reason = "Unmanaged"
 	} else if deploy == nil {
-		if e, ok := applyError.(permanentError); ok {
-			operatorAvailable.Message = applyError.Error()
-			operatorAvailable.Reason = e.Reason
-		} else if removed {
+		if removed {
 			operatorAvailable.Status = operatorapiv1.ConditionTrue
 			operatorAvailable.Message = "The registry is removed"
 			operatorAvailable.Reason = "Removed"
+		} else if e, ok := applyError.(permanentError); ok {
```

Comment:
Why do you want to reorder conditions?

And now I see we have a bug for Progressing when the state is Removed and applyError is not nil. We'll loose the error.

@ricardomaraschini can you remove this `removed` argument from the function and use `cr.Spec.ManagementState`? It should be less confusing.

Reply:
Original code restored.